src/ipc/ipc4/handler.c

```diff
@@ -1088,10 +1090,18 @@ static int ipc4_get_large_config_module_instance(struct ipc4_message_request *ip
 							     data,
 							     (const char *)MAILBOX_HOSTBOX_BASE);
 	} else {
+#if CONFIG_LIBRARY
+		/* copy the data after the reply message */
+		ret = drv->ops.get_large_config(dev, config.extension.r.large_param_id,
+						config.extension.r.init_block,
+						config.extension.r.final_block,
+						&data_offset, data + sizeof(reply));
+#else
```

Comment:
Possibly nicer would be just have local variable to calculate 'data' and not ifdef the whole get_large_config call.

Reply:
agree. fixed